### PR TITLE
[FEATURE ember-runtime-sortable-sort-order] Sort order per proeprty in Sortable mixin

### PR DESCRIPTION
**Pull request by huafu**
This is a non breaking change that would allow one to specify a different sort order per property in `sortProperties` array.

Let's say you have an array of users each of who has a weight. Weight can be the same for 2 users so then one would want the order to use the names, and logically the user with the most weight should be on the top of the list. With this feature it's possible now by adding a `:desc` at the end of the property which needs to be inverted before being included in the sort order:

``` javascript
sortProperties: ['weight:desc', 'name'],
```

Basically the `Sortable` mixin has the same behaviour as before (related to `sortProperties` and `sortAscending`) except that now you can append one or more property name(s) in `sortProperties` array with a `:desc` so that for that property(ies) the order function will work the other way around.

### CODE REVIEW

**Comment by jprberlin** [comment]
@huafu What happened to this PR? Can you re-merge master, i think this is actually a nice new feature.

**Comment by huafu** [comment]
@jprberlin well IDK, I was waiting for Ember core team to talk about it in a meeting but I guess it got forgotten. K will re-re-re-merge master lol ;-)

**Comment by huafu** [comment]
@jprberlin voila ;-)

cc @rwjblue @stefanpenner

**Comment by stefanpenner** [comment]
@huafu thanks for your work here. Unfortunately sortable mixin is now deprecated, although extracted as an addon (for legacy support) it has likely reached EOL.

Going forward, glimmer is able to efficient handle these re-render cases, so a more naive resort/filter the world strategy is both more efficient and less bugger.

Thanks again for your effort here.
